src/Application/Actions/Donations/Create.php

```diff
@@ -25,23 +26,14 @@
 
 class Create extends Action
 {
-    private DonationRepository $donationRepository;
-    private EntityManagerInterface $entityManager;
-    private SerializerInterface $serializer;
-    private StripeClient $stripeClient;
-
+    #[Pure]
     public function __construct(
-        DonationRepository $donationRepository,
-        EntityManagerInterface $entityManager,
-        LoggerInterface $logger,
-        SerializerInterface $serializer,
-        StripeClient $stripeClient
+        private DonationRepository $donationRepository,
+        private EntityManagerInterface $entityManager,
+        protected LoggerInterface $logger,
```

Comment:
I'm not sure it makes sense to re-declare `$logger`'s visibility – I don't think we should be promoting it because that's handled by the parent class through the parent constructor call.